torch/distributed/distributed_c10d.py

```diff
@@ -1526,6 +1551,83 @@ def _set_pg_timeout(timeout: timedelta, group: Optional[ProcessGroup] = None) ->
         backend._set_default_timeout(timeout)
 
 
+@_exception_logger
+@_time_logger
+def init(
```

Comment:
Should this be renamed `init_metadata` or something? Just `init()` implies it does everything rather than a restricted subset of initialization.